doc/memory

```diff
@@ -21,88 +22,84 @@ functions, including boot, filesystem access, context switching, etc.
+The kernel supports preemptive multitasking, and to this end it keeps track of
+running processes and their resources. The process table is where this state is
+stored, and it is allocated in the heap and grows or shrinks and demands change.
+There are some variables in misc kernel state that keep track of the process
+table:
 
-Each of the tables is a fixed-length array that holds several of the following
-data structures.
+    process_table       Pointer to process table in the heap
+    active_processes    Number of active processes
+    current_process     Index of the currently running process
+    last_pid            The last PID allocated by the kernel
 
-                                    Threads
+At boot, the kernel allocates a process table large enough to hold 8 processes.
+As more processes are started, the process table may be `realloc`d to hold
+more or less process state (though never less than 8). Each process state
+structure is designed like so:
 
     Address Length  Description
 
-    0x0000  0x01    Thread ID
+    0x0000  0x01    Process ID (pid)
     0x0001  0x02    Executable address
-    0x0003  0x02    Stack pointer
+    0x0003  0x02    Stack address (SP)
     0x0005  0x01    Flags (bitfield)
-                    1: May be suspended
-                    2: Is suspended
-                    3: Color enabled
-    0x0006  0x02    Reserved for future use
-
-                                   Libraries
-
-    Address Length  Description
-
-    0x0000  0x01    Library ID
-    0x0001  0x02    Library address
-    0x0003  0x01    Dependent threads
-
-                                    Signals
-
-    Address Length  Description
-
-    0x0000  0x01    Target thread ID
-    0x0001  0x01    Message type
-    0x0002  0x02    Payload
-
-                                  File Handles
+                    0: May be suspended
+                    1: Is suspended
+                    2: Color enabled (84+CSE only)
+    0x0006  0x02    Pointer to file handle table
+    0x0008  0x02    Pointer to library handle table
+    0x000A  0x02    Pointer to signal table
+    0x000C  0x03    Reserved for future use
+
+Each process has its own stack, file descriptor table, library handle table, and
+signal table. These are also allocated in the heap and expand or shrink to meet
+the process's needs. By default, each process is allocated a file descriptor
+table 4 entries long, a library handle table 2 entries long, and no signal
+table (the pointer will equal zero). Each table begins with a byte giving the
+number of entries present, then the table itself follows.
+
+                                  File handle
 
     Address Length  Description
 
-    0x0000  0x01    Flags/Owner
+    0x0000  0x01    Flags (bitfield)
+                    0: Writable if 1
+                    1: EoF if 1
+                    2: Flushed if 1
+                    3: Set if final block of the file*
+                    4-7: Reserved for future use
```

Comment:
Previously, flags were split across two different parts of the file handle entry, because I was lazy. This merges the flags into one byte (since we don't need owner any more) and rearranges them a bit with future changes in mind.